editor/components/inserter/index.js

```diff
@@ -39,12 +52,46 @@ class Inserter extends Component {
 		}
 	}
 
+	showInsertionPoint() {
+		const { showInlineInsertionPoint, showInsertionPoint } = this.props;
+
+		if ( this.isInsertingInline() ) {
+			this.setState( { isInline: true } );
+			showInlineInsertionPoint();
+		} else {
+			this.setState( { isInline: false } );
+			showInsertionPoint();
+		}
+	}
+
+	hideInsertionPoint() {
+		const { hideInlineInsertionPoint, hideInsertionPoint } = this.props;
+
+		if ( this.state.isInline ) {
+			hideInlineInsertionPoint();
+		} else {
+			hideInsertionPoint();
+		}
+	}
+
+	isInsertingInline() {
+		const { selectedBlock, canInsertInline } = this.props;
+		const isEmptyParagraph = selectedBlock &&
```

Comment:
I don't understand the paragraph check here. It's generalised to all `RichText` components?

Reply:
If you still need this logic for some reason, `isUnmodifiedDefaultBlock` might work better here.

Reply:
An empty Paragraph is a special case where the Inline Blocks menu should not render (see https://github.com/WordPress/gutenberg/pull/5794#issuecomment-376059920).

I agree, using `isUnmodifiedDefaultBlock` is a more elegant solution. Made the change in https://github.com/WordPress/gutenberg/pull/5794/commits/6bebead9db10c48e6ed531ec82b10c521feb83af.